axelrod/tests/unit/test_thuemorse.py

```diff
@@ -0,0 +1,60 @@
+"""Test for the Thue-Morse strategies."""
+
+import axelrod
+from .test_player import TestPlayer
+
+C, D = axelrod.Actions.C, axelrod.Actions.D
+
+
+class TestThueMorse(TestPlayer):
+
+    name = 'ThueMorse'
+    player = axelrod.ThueMorse
+    expected_classifier = {
+        'memory_depth': float('Inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        """Test that strategy is randomly picked (not affected by history)."""
+        self.first_play_test(D)
+        self.first_play_test(D)
+
+    def test_effect_of_strategy(self):
+        self.markov_test(['C', 'C', 'C', 'C'])
+        self.responses_test([], [], [D, C, C, D, C, D, D, C, C, D, D, C, D, C, C, D])
+        self.responses_test([C], [C], [C, C, D, C, D, D, C, C, D, D, C, D, C, C, D])
+        self.responses_test([D], [D], [C, C, D, C, D, D, C, C, D, D, C, D, C, C, D])
+        self.responses_test([C, C, C, D], [C, C, C, D], [C, D, D, C, C, D, D, C, D, C, C, D])
+
+
+
+
+class TestThueMorseInverse(TestPlayer):
+
+    name = 'ThueMorseInverse'
+    player = axelrod.ThueMorseInverse
+    expected_classifier = {
+        'memory_depth': float('Inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        """Test that strategy is randomly picked (not affected by history)."""
```

Comment:
Cool: thanks for clarifying. :)

On Mon, Feb 15, 2016 at 5:50 PM Marc Harper <notifications@github.com>
wrote:

> In axelrod/tests/unit/test_thuemorse.py
> <https://github.com/Axelrod-Python/Axelrod/pull/460#discussion_r52927203>:
>
> > +
> > +class TestThueMorseInverse(TestPlayer):
> > +
> > +    name = 'ThueMorseInverse'
> > +    player = axelrod.ThueMorseInverse
> > +    expected_classifier = {
> > +        'memory_depth': float('Inf'),
> > +        'stochastic': False,
> > +        'makes_use_of': set(),
> > +        'inspects_source': False,
> > +        'manipulates_source': False,
> > +        'manipulates_state': False
> > +    }
> > +
> > +    def test_strategy(self):
> > +        """Test that strategy is randomly picked (not affected by history)."""
>
> It's still infinite memory I think even though it doesn't use the other
> players history. Unless the sequence repeats eventually or has a finite
> order recurrence relation (doesn't seem so) then the generation of the
> sequence depends on the round_number. So it depends on the history of play
> (just the length, not the content).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/460/files#r52927203>.
>